### PR TITLE
[Merged by Bors] - chore(MetricSpace/HausdorffDistance): split in two

### PR DESCRIPTION
**Pull request by grunweg**
The file was becoming a bit large (1550 lines). Split in two files of about 900 and 700 lines:
the first file contains more basic material, the second file contains all material related to thickenings.

Extend the module docstrings by mentioning the main results in this file.

---

Feedback about the new docstrings is welcome; I wrote them from scratch.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/MetricSpace/HausdorffDistance/Basic.lean [hunk 1]
```diff
@@ -26,15 +25,34 @@ This files introduces:
...
+* `IsCompact.exists_infEdist_eq_edist`: if `s` is compact and non-empty, there exists a point `y`
+  which attains this edistance
+* `IsOpen.exists_iUnion_isClosed`: every open set `U` can be written as the increasing union
+  of countably many closed subsets of `U`
+
+* `hausdorffEdist_comm`: the Hausdorff edistance is symmetric
+* `hausdorffEdist_triangle`: the Hausdorff edistance satisfies the triangle inequality
+* ``hausdorffEdist_zero_iff_closure_eq_closure`: two sets have Hausdorff edistance zero
+  iff their closures coincide.
+* `hausdorffEdist_closure`: replacing a set by its closure does not change the Hausdorff edistance
```
Comment (grunweg) [inline Mathlib/Topology/MetricSpace/HausdorffDistance/Basic.lean:42]:
```suggestion
* in particular, closed sets in an emetric space are an emetric space (this is shown in `EMetricSpace.closeds.emetricspace`)

* `hausdorffEdist_closure`: replacing a set by its closure does not change the Hausdorff edistance

**Comment by YaelDillies** [comment]
Can you explain what the two new files contain in the PR description?

**Comment by grunweg** [comment]
Merged master, and extended the PR description.

File: Mathlib/Topology/MetricSpace/HausdorffDistance/Basic.lean [hunk 2]
```diff
@@ -26,15 +25,37 @@ This files introduces:
...
+  which attains this edistance
+* `IsOpen.exists_iUnion_isClosed`: every open set `U` can be written as the increasing union
+  of countably many closed subsets of `U`
+
+* `hausdorffEdist_comm`: the Hausdorff edistance is symmetric
+* `hausdorffEdist_triangle`: the Hausdorff edistance satisfies the triangle inequality
+* ``hausdorffEdist_zero_iff_closure_eq_closure`: two sets have Hausdorff edistance zero
+  iff their closures coincide.
+* in particular, closed sets in an emetric space are an emetric space
+  (this is shown in `EMetricSpace.closeds.emetricspace`)
```
Comment (YaelDillies) [inline Mathlib/Topology/MetricSpace/HausdorffDistance/Basic.lean:43]: That's too much stuff mentioned. Things like `hausdorffEdist_comm` can be readily found from their name.

**Review by YaelDillies — COMMENTED** [review]
I think it makes more sense to move the thickening material to `Topology.MetricSpace.Thickening`. It's not really a subset of Hausdorff distance material.

File: Mathlib/Topology/MetricSpace/HausdorffDistance/Basic.lean (hunk 2 above)
Comment (grunweg) [reply Mathlib/Topology/MetricSpace/HausdorffDistance/Basic.lean]: Fair point; I have shortened the docstring a bit.

**Comment by grunweg** [comment]
@YaelDillies Makes sense, I have moved the files as you suggested.

**Review by YaelDillies — APPROVED** [review]
maintainer merge

File: Mathlib/Topology/MetricSpace/HausdorffDistance.lean [hunk 3]
```diff
@@ -26,15 +25,35 @@ This files introduces:
...
+* `infEdist_closure`: the edistance to a set and its closure coincide
+* `EMetric.mem_closure_iff_infEdist_zero`: a point `x` belongs to the closure of `s` iff
+  `infEdist x s = 0`
+* `IsCompact.exists_infEdist_eq_edist`: if `s` is compact and non-empty, there exists a point `y`
+  which attains this edistance
+* `IsOpen.exists_iUnion_isClosed`: every open set `U` can be written as the increasing union
+  of countably many closed subsets of `U`
+
+* `hausdorffEdist_closure`: replacing a set by its closure does not change the Hausdorff edistance
+* ``hausdorffEdist_zero_iff_closure_eq_closure`: two sets have Hausdorff edistance zero
```
Comment (sgouezel) [inline Mathlib/Topology/MetricSpace/HausdorffDistance.lean:39]:
```suggestion
* `hausdorffEdist_zero_iff_closure_eq_closure`: two sets have Hausdorff edistance zero
```

**Review by sgouezel — COMMENTED** [review]
bors d+
Thanks!

File: Mathlib/Topology/MetricSpace/Thickening.lean [hunk 4]
```diff
@@ -0,0 +1,695 @@
+/-
+Copyright (c) 2019 Sébastien Gouëzel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sébastien Gouëzel
+-/
```
[the remark this answers is not in the document]
Comment (grunweg) [reply Mathlib/Topology/MetricSpace/Thickening.lean]: Good point; thank you for pointing this out!

**Comment by grunweg** [comment]
Thank you for the review.
bors r+

**Pull request closed without merging**